tests/functional/test_pre_configure.py

```diff
@@ -37,26 +36,22 @@ def envar_exporter(dict_):
 
 
 @pytest.mark.parametrize(
-    'srcdir, expect',
+    'srcdir',
     [
         param(
             '07_cli_override',
-            'CLI_VAR=="Wobble", "failed 1.1"',
             id='template variable not set'
         ),
         param(
             '08_template_engine_conflict',
-            '.*empy.*#!jinja2.*',
             id='template engine conflict'
         )
     ]
 )
-def test_validate_fail(srcdir, expect, cylc_validate_cli):
+def test_validate_fail(srcdir, cylc_validate_cli):
     srcdir = Path(__file__).parent / srcdir
-    validate = cylc_validate_cli(srcdir)
-    assert validate.ret == 1
-    if expect:
-        assert re.findall(expect, str(validate.exc))
+    result = cylc_validate_cli(str(srcdir))
+    assert result.ret != 0
```

Comment:
This seems a little complicated. Rather than running these scripts through something which emulates `Popen` then checking the artificial return code to determine whether it failed, why not run the check directly e.g:

```python
def test_validate_fail(srcdir):
    srcdir = Path(__file__).parent / srcdir
    with pytest.raises(Exception):
        validate(src_dir)
```

That way the tests can interact with the code directly, catch/expect the exceptions by name, mock/path, caplog, capsys as appropriate.